src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs

```diff
@@ -70,15 +70,7 @@ public async Task StartAsync(CancellationToken cancellationToken = default)
 
             if (_options.ServicesStartConcurrently)
             {
-                Task tasks = Task.WhenAll(_hostedServices.Select(async service =>
-                {
-                    await service.StartAsync(combinedCancellationToken).ConfigureAwait(false);
-
-                    if (service is BackgroundService backgroundService)
-                    {
-                        _ = TryExecuteBackgroundServiceAsync(backgroundService);
-                    }
-                }));
+                Task tasks = Task.WhenAll(_hostedServices.Select(service => Task.Run(() => StartAndTryToExecuteAsync(service, combinedCancellationToken), cancellationToken)));
```

Comment:
Why does this use cancellationToken rather than combinedCancellationToken as the Task.Run argument?

Also, since this isn't a hot code path I assume we're ok with the extra allocation that's incurred here for the closures and whatnot?

Reply:
> Why does this use cancellationToken rather than combinedCancellationToken as the Task.Run argument?

I fixed the combined cancellation tokens, that was an error from me.

> Also, since this isn't a hot code path I assume we're ok with the extra allocation that's incurred here for the closures and whatnot?

However I don't if eliminating the closure in Task.Run is good. I know that Task has constructor that accept Action<object?> action, object? state, CancellationToken cancellationToken. and that Task.Factory.StarNew has the same calls. However I am unable to weight the gains of reducing a closure vs the loss from a cast from object to CancellationToken.